common/state/state_kv.cpp

```diff
@@ -270,9 +270,8 @@ class pstate::kv_node {
             deserialize_next_level_ids(decrypted_buffer);
         }
 
-        pstate::StateBlockId unload(ByteArray state_encryption_key) {
-            ByteArray b = serialize_next_level_ids();
-            ByteArray baEncryptedData = pdo::crypto::skenc::EncryptMessage(state_encryption_key, b);
+        pstate::StateBlockId unload(const ByteArray& state_encryption_key) {
+            ByteArray baEncryptedData = pdo::crypto::skenc::EncryptMessage(state_encryption_key, serialize_next_level_ids());
```

Comment:
do you want this to be a reference returned? this isn't a big bytearray... but its still a copy.